api/envoy/service/mutation/v3alpha/mutation.proto

```diff
@@ -0,0 +1,275 @@
+  google.protobuf.Struct new_metadata = 6;
+
+  // Change how and if the filter waits for responses from the server
+  ProcessingModeChange async_mode_change = 7;
+}
+
+// This message is sent to the mutation server when the HTTP request is
+// first received.
+// [#next-free-field: 9]
+message Request {
+  // A unique ID that will match the one sent in the "Response" message
+  string id = 1;
+
+  // The HTTP request method, such as `GET`, `POST`.
+  string method = 2;
+
+  // The HTTP request headers. If multiple headers share the same key, they
+  // must be merged according to the HTTP spec. All header keys must be
+  // lower-cased, because HTTP header keys are case-insensitive.
+  config.core.v3.HeaderMap headers = 3;
+
+  // The request target, as it appears in the first line of the HTTP request. This includes
+  // the URL path and query-string. No decoding is performed.
+  string path = 4;
+
+  // The HTTP request `Host` or 'Authority` header value.
+  string host = 5;
+
+  // The HTTP URL scheme, such as `http` and `https`.
+  string scheme = 6;
+
+  // Information about the upstream service's connection
+  auth.v3.AttributeContext.Peer client_info = 7;
+
+  // The values of properties selected by the "request_properties"
+  // list in the configuration. Each entry in the map is the
+  // value of the specified property.
+  map<string, google.protobuf.Struct> properties = 8;
+}
+
+// This message is sent to the mutation server when the HTTP response is first
+// received.
+// [#next-free-field: 6]
+message Response {
+  // A unique ID that will match the one sent in the "Request" message
+  string id = 1;
+
+  // The HTTP status on the response
+  int32 status_code = 2;
+
+  // The HTTP headers
+  config.core.v3.HeaderMap headers = 3;
+
+  // Information about the upstream service's connection
+  auth.v3.AttributeContext.Peer destination_info = 4;
+
+  // The values of properties selected by the "response_properties"
+  // list in the configuration. Each entry in the map is the
+  // value of the specified property.
+  map<string, google.protobuf.Struct> properties = 5;
+}
+
+// This message contains the message body that Envoy sends to the mutation service.
+message Body {
+  bytes body = 1;
+
+  bool end_of_stream = 2;
+}
+
+// This message completely replaces the request or response.
+// If a STOP was also returned, then short-circuit the
+// rest of the filter processing, and if on the request path,
+// return this response directly back to the client.
+//
+// Otherwise, replace everything
+// and let other filters run as they would like.
+message CompleteResponse {
+  type.v3.HttpStatus status = 1 [(validate.rules).message = {required: true}];
+
+  repeated config.core.v3.HeaderValue headers = 2;
+
+  bytes body = 3;
+
+  repeated config.core.v3.HeaderValue trailers = 4;
+}
+
+// Change all or part of the message body by replacing the portion
+// of the original body denoted by "offset" and "length" with "body."
```

Comment:
Is this relative to the start of the body or the start of the last chunk?

Reply:
I assume last chunk?